### ISSUE TITLE
Cubic#transform() doesn't guarantee 0 for t=0 or 1 for t=1

### ISSUE THREAD

**Issue by marcoms** [flutter/flutter#27406]
According to `Curve` docs:

```dart
/// A curve must map t=0.0 to 0.0 and t=1.0 to 1.0.
```

Running a simple test with the following gives an undesired result:

```dart
print(Cubic(0, 0, 1, 1).transform(0));  // -> 0.00072479248046875
print(Cubic(0, 0, 1, 1).transform(1));  // -> 0.9992752075195312
```

**Comment by marcoms** [comment]
Also occurring in Elastic curves

**Comment by marcoms** [comment]
Fixed by #27409
